scheduler/src/test/java/io/camunda/zeebe/scheduler/retry/RetryStrategyTest.java

```diff
@@ -116,6 +118,25 @@ public void shouldAbortOnOtherException() {
     assertThat(resultFuture.getException()).isExactlyInstanceOf(RuntimeException.class);
   }
 
+  @Test
+  public void shouldNotInterleaveRetry() {
+    // given
+    final AtomicReference<ActorFuture<Boolean>> firstFuture = new AtomicReference<>();
```

Comment:
If you revert the fix and use `submit` again, you will see this test fails (the first future is incomplete), and you will get the logs that the second future was completed twice.